### PR TITLE
AMI-1663 read userdata

### PR DESCRIPTION
**Pull request by MatthaeusHarris**
PLEASEREVIEW @declanshanaghy @cloudbuilder @cgaspardSplunk @nsamad 

Incorporates the get_user_data function recently merged into splunk_cloud_common_node to fetch the STACK_NAME from instance userdata

Also, I fixed some tests that were broken.

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -3,7 +3,12 @@
 var os = require('os');
 var fs = require('fs');
 var sidecar = require('../lib/eureka-sidecar');
-require('longjohn');
+var SplunkCloud = require('splunk_cloud_common_node');
+var SplunkCloudDynamicConfig = new SplunkCloud.DynamicConfig();
+var winston = SplunkCloud.Logger;
+if (process.env.DEBUG) {
+	require('longjohn');
```
Comment (nsamad) [inline bin/cli.js]: So, we're making longjohn configurable based on an env var?

File: bin/cli.js [hunk 2]
```diff
@@ -12,17 +17,20 @@ fs.readFile('/etc/buildnumber', function(err, data) {
...
+		sidecar.initialize({
+			'cloud_stack': process.env.CLOUD_STACK,
+			'app': process.env.APP || 'sidecar',
+			'ip-address': process.env.IP_ADDRESS || os.networkInterfaces().eth0[0].address,
+			'app-port': process.env.PORT || '5000',
+			'proto': process.env.PROTO || 'http',
+			'hostname': process.env.HOSTNAME || os.hostname(),
+			'metadata': {
+				'version': process.env.VERSION || '0.0.1',
+				'build': buildNumber
```
Comment (nsamad) [inline bin/cli.js]: What version is this?

File: bin/cli.js (hunk 1 above)
Comment (MatthaeusHarris) [inline bin/cli.js]: Yes.  In sidecar, there aren't any sensitive data to leak, so it doesn't need to be turned off.  However, it's good practice to leave it off except in debug situations.

File: bin/cli.js (hunk 2 above)
Comment (MatthaeusHarris) [inline bin/cli.js]: From previously in the file:  buildNumber is read from /etc/buildnumber (which aminator writes).  Version is the version of the microservice, as specified in the environment variable set by supervisor.

Comment (declanshanaghy) [inline bin/cli.js]: Sidecar should be reporting the version from the health endpoint.

Comment (MatthaeusHarris) [inline bin/cli.js]: https://jira.splunk.com/browse/AMI-833 This feature was triaged out months ago.  Do you want me to add it in now?

**Comment by MatthaeusHarris** [comment]
Verbal :zap: from @declanshanaghy, moving on to security review.

**Pull request merged**